integration/test/basic/basic_test.go

```diff
@@ -4,14 +4,62 @@ package basic
 
 import (
 	"context"
+	"fmt"
 	"testing"
+	"time"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func TestHelm(t *testing.T) {
+// TestBasic ensures that there is a ready coredns-app deployment.
+func TestBasic(t *testing.T) {
+	ctx := context.Background()
+	var err error
+
+	// Check coredns-app deployment is ready.
+	err = checkReadyDeployment(ctx)
+	if err != nil {
+		t.Fatalf("could not get coredns-app: %v", err)
+	}
+}
+
+func checkReadyDeployment(ctx context.Context) error {
 	var err error
 
-	err = ba.Test(context.Background())
+	l.LogCtx(ctx, "level", "debug", "message", "waiting for ready deployment")
+
+	o := func() error {
+		selector := fmt.Sprintf("%s=%s", "app.kubernetes.io/name", app)
+		lo := metav1.ListOptions{
+			LabelSelector: selector,
+		}
+
+		deploys, err := appTest.K8sClient().AppsV1().Deployments(testNamespace).List(ctx, lo)
+		if err != nil {
+			return microerror.Mask(err)
+		} else if len(deploys.Items) == 0 {
+			return microerror.Maskf(executionFailedError, "deployment with label%#q in %#q not found", app, testNamespace)
+		}
+
+		deploy := deploys.Items[0]
+
+		if deploy.Status.ReadyReplicas != *deploy.Spec.Replicas {
+			return microerror.Maskf(executionFailedError, "deployment %#q want %d replicas %d ready", deploy.Name, *deploy.Spec.Replicas, deploy.Status.ReadyReplicas)
+		}
+
+		return nil
+	}
+	b := backoff.NewConstant(2*time.Minute, 5*time.Second)
+	n := backoff.NewNotifier(l, ctx)
+
+	err = backoff.RetryNotify(o, b, n)
 	if err != nil {
-		t.Fatalf("%#v", err)
+		return microerror.Mask(err)
```

Comment:
Why microerror instead of testing here?

Reply:
In this case there was an error since func o return a non nil err value.

What test would you run?

Reply:
The `t.Fatalf()` comes from the `testing` library, whereas the `microerror.Mask()` is from our own `microerror`. I see now that `TestBasic()` handles this microerror return and uses the `t.Fatalf()` so that the test suite knows the test failed. So all good here :+1:

Reply:
@stone-z I was not aware of that. Thank you for the explanation :)